axum-extra/src/extract/query.rs

```diff
@@ -216,11 +216,9 @@ impl std::error::Error for QueryRejection {
 ///
 /// [example]: https://github.com/tokio-rs/axum/blob/main/examples/query-params-with-empty-strings/src/main.rs
 #[cfg_attr(docsrs, doc(cfg(feature = "query")))]
-#[deprecated = "Use Option<Query<_>> instead"]
```

Comment:
While I don't think people should use `Option<Query<_>>`, should we recommend using `Query<_>` with `Option<_>` fields instead and keep this deprecated?

Reply:
Well this extractor was specifically added with the intention of allowing "all query parameters specified, or none". I don't really get why that's a useful thing, but we can always deprecate / remove this later.

Reply:
When you say deprecate, we cannot lint that through the attribute though, right?

Reply:
Huh, why would I not mean the attribute?

Reply:
Note that this is about `OptionalQuery` in axum-extra, not the `OptionalFromRequestParts` impl for `Query` (already removed in this PR).

Reply:
Sorry, I got confused about what the deprecation was about.